extensions/amp-link-rewriter/amp-link-rewriter.md

```diff
@@ -0,0 +1,218 @@
+The final code should look like:
+
+```html
+<!doctype html>
+<html ⚡>
+<head>
+  ...
+  <script async custom-element="amp-link-rewriter" src="https://cdn.ampproject.org/v0/amp-link-rewriter-0.1.js"></script>
+  ...
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+    ...
+    <amp-link-rewriter
+        layout="nodisplay">
+
+        <script type="application/json">
+                {
+                    "output": "https://visit.digidip.net?pid=110&url=${href}&cid=${customerId}",
+                    "section": [
+                        "#product-listing-1",
+                        "#product-listing-2",
+                    ],
+                    "attribute": {
+                        "href": "`((?!(https:\/\/youtube\.com)|(https:\/\/mobile\.vodafone\.de)).)*`",
+                        "id": "comments",
+                        "class": "sidebar",
+                        "rel": "(?!(skip))*",
+                    },
+                    "vars": {
+                        "customerId": "12345"
+                    }
+                }
+         </script>
+
+    </amp-link-rewriter>
+    ....
+</body>
+</html>
+```
+
+## JSON configuration
+
+##### output (required)
+
+The "output" property is the redirection url plus a query string of placeholders that will be shifted with values defined in the config JSON 'vars' property, or in the anchor itself as a data attribute.
+
+Example:
+```html
+    <amp-link-rewriter
+        layout="nodisplay">
+
+        <script type="application/json">
+                {
+                    "output": "https://visit.digidip.net?pid=110&cid=${customerId}`",
+                    "vars": {
+                        "customerId": "12345"
+                    }
+                }
+         </script>
+
+    </amp-link-rewriter>
+```
+
+We can also define data values in the anchor data attribute as the following example:
+```html
+    <a href="https://www.amazon.de/s?i=computers&rh=n%3A340843031&ref=sn_gfs_co_computervs_AM_5" data-vars-event-id="231">
+```
+
+The config could be something like:
+
+```json
+    {
+      "output": "https://visit.digidip.net?eid=${eventId}&cid=${customerId}"
+
+    }
+```
+The resulting, rewritten URL would be:
+```url
+`https://visit.digidip.net?eid=231&cid=12345`
+```
+
+Besides defined placeholders that match the data defined in the 'vars' property of the JSON configuration, or as a data attribute, there are other pre-defined placeholders that will be shifted with information such as anchor url, page location, referrer page, or anchor id. The following table shows the relationship between defined data and placeholders.
+
+| value          | source     |       example                                         |    placeholder       
+| -------------- | ---------- |-------------------------------------------------------|----------------------
+| location       | page       |    `https://www.pepper.com/`                          |  `${location}`        
```

Comment:
let's add one more useful var: `${random}`. it can just return a random string from `Math.random().toString(32).substr(2)`